lib/figleaf/config.rb

```diff
@@ -0,0 +1,53 @@
+module Figleaf
+  # Convert a ruby block to nested hash
+  class Config
+    def initialize
+      @property = {}
+    end
+
+    def self.define(property_name, &block)
+      property = new.define(&block)
+
+      Settings.configure_with_auto_define do
+        if Settings.respond_to?(property_name) &&
```

Comment:
Infer name from file name if file location is `config/settings`. Otherwise, require property name to be passed in.